pyEpiabm/pyEpiabm/simulation.py

```diff
@@ -0,0 +1,94 @@
+#
+# Simulates a complete pandemic
+#
+import os
+import typing
+from .abstract_sweep import AbstractSweep
+from .infection_status import InfectionStatus
+from ._csv_dict_writer import _CsvDictWriter
+from .population import Population
+
+
+class Simulation:
+    """Class to run a full simulation.
+    """
+    def configure(self,
+                  population: Population,
+                  initial_sweeps: typing.List[AbstractSweep],
+                  sweeps: typing.List[AbstractSweep],
+                  sim_params: typing.Dict,
+                  file_params: typing.Dict):
+        """Initialise a population structure for use in the simulation.
+
+        :param population: Population structure for the model
+        :type population: Population
+        :param pop_params: Dictionary of parameter specific to the population
+        :type pop_params: dict
+        :param initial_sweeps: List of abstract sweeps used to initialise the
+            simulation
+        :type initial_sweeps: list
+        :param sweeps: List of abstract sweeps used in the simulation. Queue
+            sweep and host progression sweep must appear at the end of the
+            list
+        :type sweeps: list
+        :param sim_params: Dictionary of parameters specific to the simulation
+            used
+        :type sim_params: dict
+        :param file_params: Dictionary of parameters specific to the output
+            file
+        :type file_params: dict
+        """
+        self.sim_params = sim_params
+        self.population = population
+        self.initial_sweeps = initial_sweeps
+        self.sweeps = sweeps
+        # Initial sweeps configure the population by changing the type,
+        # infection status, infectiveness or susceptibility of people
+        # or places. Only runs on the first timestep.
+        for s in initial_sweeps + sweeps:
+            assert isinstance(s, AbstractSweep)
```

Comment:
```suggestion
```
python doesn't care about the type, just that it has a bind_population method. Unless we were writing our own raise error messages, I'd do without this and let it fail later if the needed methods are not implemented